packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -1086,6 +1089,26 @@ async function handler(
       },
       loggerWithEventDetails
     );
+    if (req.body.allRecurringDates && req.body.allRecurringDates.length > 1) {
+      for (
+        let i = 1;
+        i < req.body.allRecurringDates.length && i < req.body.numSlotsToCheckForAvailability;
+        i++
+      ) {
+        const start = req.body.allRecurringDates[i].start;
+        const end = req.body.allRecurringDates[i].end;
+        await ensureAvailableUsers(
+          eventTypeWithUsers,
+          {
+            dateFrom: dayjs(start).tz(reqBody.timeZone).format(),
+            dateTo: dayjs(end).tz(reqBody.timeZone).format(),
+            timeZone: reqBody.timeZone,
+            originalRescheduledBooking,
+          },
+          loggerWithEventDetails
+        );
+      }
+    }
```

Comment:
Now that we are checking for the availability of all slots(as per numSlotsToCheckForAvailability) on first slot iteration itself, what's the need of doing `ensureAvailableUsers` on line 1082 for recurring event.

We could make the code easier to understand by having `isFirstSlot` property in req.body and do away with `skipAvailabilityCheck`.

```javascript
if(req.body.allRecurringDates) {
	if (req.body.isFirstSlot) {
	  // This code to check for availability for all required slots(as per `numSlotsToCheckForAvailability`)
	}
} else {
 // Line 1082 Code
}
```

Reply:
This logic is faulty in current state. If we set numSlotsToCheckForAvailability to 3 or more, a recurring event can never be booked because for 3rd slot we would check for 2nd slot availability as well which will always be booked because we reached 3 after iterating 2.

Reply:
@hariombalhara 

> Blocker: This logic is faulty in current state. If we set numSlotsToCheckForAvailability to 3 or more, a recurring event can never be booked because for 3rd slot we would check for 2nd slot availability as well which will always be booked because we reached 3 after iterating 2.

After the first call to `handleNewBooking` in `recurring-event.ts`, `thirdPartyRecurringEventId` will be set. For every subsequent call to `handleNewBooking`, `skipAvailabilityCheck` will be set to `True`. Therefore, in this example, for 3rd slot, 2nd slot availability will not be checked.

Reply:
> Now that we are checking for the availability of all slots(as per numSlotsToCheckForAvailability) on first slot iteration itself, what's the need of doing `ensureAvailableUsers` on line 1082 for recurring event.
> 
> We could make the code easier to understand by having `isFirstRecurringSlot` property in req.body, and do away with `skipAvailabilityCheck`.
> 
> ```js
> if(req.body.allRecurringDates) {
> 	if (req.body.isFirstRecurringSlot) {
> 	  // This code to check for availability for all required slots(as per `numSlotsToCheckForAvailability`)
> 	}
> } else {
>  // Line 1082 Code
> }
> ```
> 
> In this way, I believe, we would be able to also remove the newly added thirdPartyRecurringEventId related logic in recurring-event.ts.

`thirdPartyRecurringEventId` would still be necessary because for every event instance within `cal.com` this value would need to be stored in the database.

`skipAvailabilityCheck` would also still be necessary for the logic that uses `numSlotsToCheckForAvailability`.

I can rewrite this code in the format you had suggested with the addition of the `isFirstRecurringSlot`. Let me know if you'd still like me to do that.

Reply:
> @hariombalhara
> 
> > Blocker: This logic is faulty in current state. If we set numSlotsToCheckForAvailability to 3 or more, a recurring event can never be booked because for 3rd slot we would check for 2nd slot availability as well which will always be booked because we reached 3 after iterating 2.
> 
> After the first call to `handleNewBooking` in `recurring-event.ts`, `thirdPartyRecurringEventId` will be set. For every subsequent call to `handleNewBooking`, `skipAvailabilityCheck` will be set to `True`. Therefore, in this example, for 3rd slot, 2nd slot availability will not be checked.

The issue comes when there are no calendars and conferencing apps connected. In that case, skipAvailability isn't correctly set. See https://www.loom.com/share/4b17b060bcab4ba0a1ec35dea8f973d7

Reply:
> thirdPartyRecurringEventId would still be necessary because for every event instance within cal.com this value would need to be stored in the database.

I meant to remove it's usage in recurring-event.ts only, the goal being simplifying the code and making it easy to understand. I still like the `isFirstRecurringSlot` approach if that's fine by you.